integration-cli/docker_cli_run_test.go

```diff
@@ -2028,10 +2028,10 @@ func (s *DockerSuite) TestRunSetMacAddress(c *check.C) {
 	mac := "12:34:56:78:9a:bc"
 	var out string
 	if daemonPlatform == "windows" {
-		out, _ = dockerCmd(c, "run", "-i", "--rm", fmt.Sprintf("--mac-address=%s", mac), "busybox", "sh", "-c", "ipconfig /all | grep 'Physical Address' | awk '{print $12}'")
+		out, _ = dockerCmd(c, "run", "-i", fmt.Sprintf("--mac-address=%s", mac), "busybox", "sh", "-c", "ipconfig /all | grep 'Physical Address' | awk '{print $12}'")
 		mac = strings.Replace(strings.ToUpper(mac), ":", "-", -1) // To Windows-style MACs
 	} else {
-		out, _ = dockerCmd(c, "run", "-i", "--rm", fmt.Sprintf("--mac-address=%s", mac), "busybox", "/bin/sh", "-c", "ip link show eth0 | tail -1 | awk '{print $2}'")
```

Comment:
`win2lin` failed on this very often, it looks like impacted by this change, but I can't think of why it can be impacted by this. Other integration tests all works fine on this.

And I checked this test case, the `--rm` is quite unnecessary, so I removed it.

Do you think we should change it back,  and do some investigation?